torch/_numpy/_funcs_impl.py

```diff
@@ -305,28 +307,29 @@ def linspace(
 
 
 def geomspace(
-    start: ArrayLike,
-    stop: ArrayLike,
+    start: ArrayLikeOrScalar,
+    stop: ArrayLikeOrScalar,
     num=50,
     endpoint=True,
     dtype: Optional[DTypeLike] = None,
     axis=0,
 ):
     if axis != 0 or not endpoint:
         raise NotImplementedError
-    base = torch.pow(stop / start, 1.0 / (num - 1))
-    logbase = torch.log(base)
+
+    # XXX: torch.logspace only accepts scalars,
+    # while np.logspace broadcasts array-like arguments
+    logbase = math.log(stop / start) / (num - 1)
```

Comment:
This won't work for arrays, so why is it now marked as ArrayLikeOrScalar?

Reply:
This never worked for arrays.
Previously: array_like arguments raise TypeError in `torch.logspace` below. (`torch.logspace` only accepts floats)
This PR: array_like arguments raise TypeError in `math.log`.

So, it's even the type of same error.

Possible alternatives:
1. introduce the whole new Scalar annotation. Won't help much since numpy does accept array-likes.
2. detect tensors with `.numel() > 1`, raise NotImplementedError ()
3. Keep as is, add to the list of low-prio things to fix _some day_ (tm).

IMO 1) and 2) is way too much, so this PR opts for the third option.

Reply:
I think it's best to simply tag it as nothing, as it was, as we shouldn't perform any wrapping logic on it.

Reply:
As before is `ArrayLike`. So on main it's converting scalars to 0D tensors for `torch.logspace` to unwrap internally (+a guard in dynamo etc). 
Do you want me to just remove all annotations from `start` and `stop`?

Reply:
I believe that'd be best, yeah. No need to wrap and unwrap when we are just dealing with scalar inputs and not tensors.

Reply:
Done in the last commit